packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp

```diff
@@ -0,0 +1,337 @@
+
+      init_task_request.set_last_task(true);
+      *last_task_request.mutable_init_task() = std::move(init_task_request);
+
+      requests.push_back(std::move(last_task_request));
+    }
+
+    return requests;
+  });
+}
+
+/**
+ * @brief Asynchronously create tasks.
+ *
+ * @param session_id The session ID.
+ * @param task_options The TaskOptions object.
+ * @param task_requests A vector of TaskRequest objects.
+ * @return A future containing a CreateTaskReply object.
+ */
+std::future<CreateTaskReply> SubmitterClient::create_tasks_async(std::string session_id, TaskOptions task_options,
+                                                                 const std::vector<TaskRequest> &task_requests) {
+  return std::async(std::launch::async, [this, task_requests, session_id = std::move(session_id),
+                                         task_options = std::move(task_options)]() mutable {
+    armonik::api::grpc::v1::Configuration config_response;
+    ClientContext context_configuration;
+
+    const auto config_status =
+        stub_->GetServiceConfiguration(&context_configuration, armonik::api::grpc::v1::Empty(), &config_response);
+    size_t chunk = 0;
+    if (config_status.ok()) {
+      chunk = config_response.data_chunk_max_size();
+    } else {
+      throw std::runtime_error("Fail to get service configuration");
+    }
+
+    CreateTaskReply reply{};
+
+    reply.set_allocated_creation_status_list(new CreateTaskReply_CreationStatusList());
+    ClientContext context_client_writer;
+    std::unique_ptr stream(stub_->CreateLargeTasks(&context_client_writer, &reply));
+
+    // task_chunk_stream(task_requests, )
+    std::vector<std::future<CreateLargeTaskRequest>> async_task_requests;
+    std::vector<std::future<CreateLargeTaskRequest>> create_large_task_requests;
+    auto create_task_request_async =
+        to_request_stream(task_requests, std::move(session_id), std::move(task_options), chunk);
+
+    for (auto &f : create_task_request_async) {
+      for (auto &create_large_task_request : f.get()) {
+        stream->Write(create_large_task_request);
+      }
+    }
+
+    stream->WritesDone();
+    Status status = stream->Finish();
+    if (!status.ok()) {
+      std::stringstream message;
+      message << "Error: " << status.error_code() << ": " << status.error_message()
+              << ". details : " << status.error_details() << std::endl;
+      throw std::runtime_error(message.str().c_str());
+    }
+
+    return reply;
+  });
+}
+
+/**
+ * @brief Submit tasks with dependencies.
+ *
+ * @param session_id The session ID.
+ * @param task_options The task options
+ * @param payloads_with_dependencies A vector of tuples containing task payload
+ * and its dependencies.
+ * @param max_retries Maximum number of retries.
+ * @return A vector of task IDs.
+ */
+std::tuple<std::vector<std::string>, std::vector<std::string>>
+SubmitterClient::submit_tasks_with_dependencies(std::string session_id, TaskOptions task_options,
```

Comment:
So only 1 result per task, should be ok for now but it's more ExtC# like than Api like imo. Definitely something to add to TODO

Reply:
Yes will do !